pandas/tests/series/indexing/test_boolean.py

```diff
@@ -168,14 +175,13 @@ def test_where_unsafe_upcast(dtype):
 @pytest.mark.parametrize("dtype", [
     np.int8, np.int16, np.int32, np.float32
 ])
-def test_where_unsafe_itemsize_fail(dtype):
-    # Can't do these, as we are forced to change the
-    # item size of the input to something we cannot.
+def test_where_upcast(dtype):
+    # see gh-9743
     s = Series(np.arange(10), dtype=dtype)
     mask = s < 5
 
     values = [2.5, 3.5, 4.5, 5.5, 6.5]
-    pytest.raises(Exception, s.__setitem__, tuple(mask), values)
+    s[mask] = values
```

Comment:
so this should still raise if you try to set with a tuple though

Reply:
a tuple will give a `ValueError: Can only tuple-index with a MultiIndex` which is tested in `series/test_indexing.py`,

the `tuple` keyword will create a tuple from an iterator, in this case a boolean series.

```python
>>> import numpy as np
>>> dtype = np.int8
>>> from pandas import Series
>>> s = Series(np.arange(10), dtype=dtype)
>>> mask = s < 5
>>> print(mask)
0     True
1     True
2     True
3     True
4     True
5    False
6    False
7    False
8    False
9    False
dtype: bool
>>> tuple(mask)
(True, True, True, True, True, False, False, False, False, False)
>>> values = [2.5, 3.5, 4.5, 5.5, 6.5]
>>> s[tuple(mask)] = values
```

raises

```python-traceback
KeyError: "None of [Index([True, True, True, True, True, False, False, False, False, False], dtype='object')] are in the [index]"
```

should we test for this?

Reply:
I think the intention was to pass a tuple of values (not tupleify one), e.g.
``s[tuple(mask, )] = values``

Reply:
in order to keep this PR more atomic should i revert the changes to this test and raise an separate issue?

Reply:
you can if you want or just fix it here, should be straightforward

Reply:
if the original intention was to raise a `Exception: cannot change dtype of input to smaller size` and that is no longer a restriction then the test is now fixed?

Reply:
@jreback 

should have said, ci failure is unrelated.. `##[error]doc/source/comparison_with_sas.rst(754,-22): error F821: undefined name 'get_ipython'`

Reply:
yes #24756 will address that

Reply:
@jreback 

> what tests catches this?

did you see my comments re #3292 and #9743

Reply:
ok i c, so this was in essence an old test.